include/event2/dns.h

```diff
@@ -782,13 +809,19 @@ struct evdns_getaddrinfo_request;
  * When the callback is invoked, we pass as its first argument the error code
  * that getaddrinfo would return (or 0 for no error).  As its second argument,
  * we pass the evutil_addrinfo structures we found (or NULL on error).  We
- * pass 'arg' as the third argument.
+ * pass 'arg' as the third argument. If a request is cached with (CNAME when
+ * required), but the cached entries are of a different address type (e.g.
+ * PF_INET vs PF_INET6), the address info is wra
  *
  * Limitations:
  *
  * - The AI_V4MAPPED and AI_ALL flags are not currently implemented.
  * - For ai_socktype, we only handle SOCKTYPE_STREAM, SOCKTYPE_UDP, and 0.
  * - For ai_protocol, we only handle IPPROTO_TCP, IPPROTO_UDP, and 0.
+ * - If we cached a response exclusively for a different address type (e.g.
+ *   PF_INET), we will set addrinfo to NULL (e.g. queried with PF_INET6)
+ * - Cache isn't hit when AI_CANONNAME is set but cached server response
+ *	 doesn't contain CNAME.
```

Comment:
Let's also reflect above that in case of cache hit it will return NULL as well

Reply:
I'm not a writer but I think this is resolved now